rds_to_s3/etl.py

```diff
@@ -0,0 +1,199 @@
+import boto3
+from botocore.exceptions import NoCredentialsError, ClientError
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(levelname)s - %(message)s",
+    handlers=[
+        logging.FileHandler("app.log"),
+        logging.StreamHandler()
+    ]
+)
+
+
+def setup_connection() -> None:
+    """Sets up a connection to the RDS"""
+    try:
+        conn = psycopg2.connect(
+            user=ENV["DB_USERNAME"],
+            password=ENV["DB_PASSWORD"],
+            host=ENV["DB_HOST"],
+            port=ENV["DB_PORT"],
+            database=ENV["DB_NAME"]
+        )
+        cursor = conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor)
+        cursor.execute(f"SET SEARCH_PATH TO {ENV['SCHEMA_NAME']};")
+        return conn
+    except OperationalError as oe:
+        logging.error(
+            "Operational error while connecting to the database: %s", oe)
+        raise
+    except InterfaceError as ie:
+        logging.error(
+            "Interface error while connecting to the database: %s", ie)
+        raise
+    except DatabaseError as de:
+        logging.error("Database error occurred: %s", de)
+        raise
+    except Exception as e:
+        logging.error(
+            "Unexpected error while setting up the database connection: %s", e)
+        raise
+
+
+def create_query_list() -> list[str]:
+    """Makes a list of the queries to extract data from the RDS"""
+    schema_name = ENV["SCHEMA_NAME"]
+    return [
+        f"SELECT u.user_id, u.first_name, u.last_name, u.phone_number FROM {schema_name}.user u ORDER BY u.user_id ASC;",
+        f"""
+        SELECT
+            s.subscription_id, s.user_id, s.subscription_status,
+            s.notification_threshold, s.keywords_id
+        FROM
+            {schema_name}.subscription s
+        JOIN
+            {schema_name}.user u ON s.user_id = u.user_id
+        ORDER BY u.user_id ASC;
+        """,
+        f"SELECT k.keywords_id, k.keyword FROM {schema_name}.keywords k ORDER BY k.keywords_id ASC;",
+        f"SELECT rt.related_term_id, rt.related_term FROM {schema_name}.related_terms rt ORDER BY rt.related_term_id ASC;",
+        f"""
+        SELECT
+            rta.related_term_assignment, rta.keywords_id, rta.related_term_id
+        FROM
+            {schema_name}.related_term_assignment rta
+        ORDER BY rta.related_term_assignment ASC;
+        """,
+        f"""
+        SELECT
+            kr.keyword_recordings_id, kr.keywords_id, kr.total_mentions,
+            kr.hour_of_day, kr.avg_sentiment
+        FROM
+            {schema_name}.keyword_recordings kr
+        ORDER BY kr.keyword_recordings_id ASC;
+        """
+    ]
+
+
+def download_csv_from_s3(bucket_name: str, file_name: str) -> pd.DataFrame:
+    """Downloads the current archive csv from S3"""
+    s3 = boto3.client(
+        "s3",
+        aws_access_key_id=ENV["AWS_ACCESS_KEY_ID"],
+        aws_secret_access_key=ENV["AWS_SECRET_ACCESS_KEY"]
+    )
```

Comment:
this could be extracted into a new function as it is repeated
```suggestion
def s3_connection() -> boto3.client:
    """Function connects to S3 and provides client to interact with it."""
    return boto3.client(
        "s3",
        aws_access_key_id=ENV["AWS_ACCESS_KEY_ID"],
        aws_secret_access_key=ENV["AWS_SECRET_ACCESS_KEY"]
    )
```